### ISSUE TITLE
SketchUp crashes on quit when using Sketchup.send_action in html dialog on_close event

### ISSUE THREAD

**Issue by valdemar** [SketchUp/api-issue-tracker#958]
Tested on SketchUp 2024 on Windows. 

SketchUp crashes when the user quits if a `HtmlDialog` is open that has a `on_closed` event handler that calls `Sketchup.send_action`

Paste the following code snippet into the ruby console and then quit SketchUp to reproduce:

```ruby
module TestingDialog

  def self.show
    @dialog = UI::HtmlDialog.new(
      dialog_title: 'SketchUp Crasher',
      style: UI::HtmlDialog::STYLE_DIALOG
    )
    @dialog.set_on_closed do
      Sketchup.send_action('selectSelectionTool:')
    end
    @dialog.show
  end

end

TestingDialog.show
```

**Comment by thomthom** [comment]
Did you test on any earlier versions?

**Comment by valdemar** [comment]
> Did you test on any earlier versions?

No, not yet. I only have SketchUp 2024 installed at the moment. 

I discovered that a solution is to use `Sketchup.active_model.select_tool(nil)` instead of `send_action` - that does not cause SketchUp to crash.

**Comment by thomthom** [comment]
> > Did you test on any earlier versions?
> 
> No, not yet. I only have SketchUp 2024 installed at the moment.
> 
> I discovered that a solution is to use `Sketchup.active_model.select_tool(nil)` instead of `send_action` - that does not cause SketchUp to crash.

Thanks for the follow up.

I'd strongly recommend avoiding `send_action` whenever possible. It's not a stable API and it'll likely be deprecated.

**Comment by kalpana-ghodeswar** [comment]
This issue has been fixed as part of release SU2026.0, hence closing.

**Comment by DanRathbun** [comment]
@kalpana-ghodeswar What was the fix? Are send actions now no-ops in dialog `#set_can_close` or `#set_on_closed` callback blocks?

**Comment by DavidMan20** [comment]
@DanRathbun The problem stemmed from the dialog's close handler. The way the code was laid out before created an order of execution error in which set_on_closed was being called after memory was freed. These weren't turned into no-ops; fixing that error solved the issue and restored intended functionality.

**Comment by DanRathbun** [comment]
David, are you part of the Extensibility Team? My query to Kalpana is a attempted nudge to **please** give more information when closing issues and **include** the notice and description of fixes in the API Release Notes.

Just checked the 2026.0 API Release Notes and this fix is **not listed**.

> The way the code was laid out before created an order of execution error in which `set_on_closed` was being called after memory was freed.

Probably using asynchronous `send_action` calls was not envisioned when the dialog class was implemented. These send actions were really intended for simple toolbar button commands.

**Comment by kalpana-ghodeswar** [comment]
Thank you, Dan. I'll try to to include the additonal info moving forward. As for the release notes, I don't see it either. I'll add a note to the ticket to ensure it's included in the next release.

**Comment by DavidMan20** [comment]
> David, are you part of the Extensibility Team? My query to Kalpana is a attempted nudge to **please** give more information when closing issues and **include** the notice and description of fixes in the API Release Notes.

Yes I am a newcomer to the extensibility team. 

> Probably using asynchronous `send_action` calls was not envisioned when the dialog class was implemented. These send actions were really intended for simple toolbar button commands.

As you said, they were meant for simple toolbar commands. I believe there were plans to deprecate send_action for a better alternative, but it’s still around because it’s still used so much by the community.

**Comment by DanRathbun** [comment]
> Yes I am a newcomer to the extensibility team.

Welcome to the jungle! ;)

**Comment by DavidMan20** [comment]
Thank you:) it's been nice so far.
